initialize2.c

```diff
@@ -6,12 +6,28 @@
 /*   By: mkimdil <mkimdil@student.42.fr>            +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/07/15 05:54:33 by mkimdil           #+#    #+#             */
-/*   Updated: 2024/07/15 05:56:35 by mkimdil          ###   ########.fr       */
+/*   Updated: 2024/08/10 00:58:48 by mkimdil          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "execute.h"
 
+void	free_arr(int **fds)
+{
+	int	i;
+
+	if (fds)
+	{
+		i = 0;
+		while (fds[i])
+		{
+			free(fds[i]);
+			i++;
```

Comment:
After freeing memory, it's a good practice to set the pointer to `NULL` to avoid potential dangling pointer issues.
And also don't forget it when you finish this while loop as well.

Also this function can use early return, I would rather do it like this:
```c
void free_arr(int **fds)
{
    int i = 0;

    if (fds == NULL)
        return;

    while (fds[i] != NULL)
    {
        free(fds[i]);
        fds[i] = NULL;  // Avoid dangling pointers
        i++;
    }
    free(fds);
    fds = NULL;  // Avoid dangling pointers
}
```